app/src/main/java/xyz/jhughes/laundry/adapters/MachineAdapter.java

```diff
@@ -135,24 +145,134 @@ public void onClick(DialogInterface dialog, int which) {
             } else {
-                fireNotificationInFuture(millisInFuture, notificationKey);
+                mContext.startService(new Intent(mContext, NotificationCreator.class)
+                        .putExtra("machine", notificationKey)
+                        .putExtra("time", milliInFuture));
+                Toast.makeText(mContext, mContext.getString(R.string.alarm_set), Toast.LENGTH_SHORT).show();
             }
-        } catch (NumberFormatException | StringIndexOutOfBoundsException e) {
-            if (m.getStatus().compareTo("Out of order") != 0) {
-                listener.postSnackbar(mContext.getString(R.string.machine_not_running), Snackbar.LENGTH_SHORT);
-            } else {
-                listener.postSnackbar("This machine is offline but may still be functioning. Visit " + m.getName() + " for details.", Snackbar.LENGTH_LONG);
+            return true;
+        } catch (NumberFormatException | StringIndexOutOfBoundsException e){
+            return false;
+        }
+
+    }
+
+    public void waitForMachine(Machine m){
+        //Constructs the dialog to wait for a machine
+        //checks the server while the dialog is open and the app is running in the background
+        final Machine m2 = m;
+        final Handler handler = new Handler();
+        lockListener.onLock();
+        AlertDialog.Builder machineWaitingDialog = new AlertDialog.Builder(mContext);
+        machineWaitingDialog.setTitle(mContext.getString(R.string.alarm))
+                .setMessage(mContext.getString(R.string.available_timer_message1) + " " + m.getName() + " " + mContext.getString(R.string.available_timer_message2))
+                .setCancelable(true)
+                .setPositiveButton(mContext.getString(R.string.available_timer_refresh), null)
+                .setNegativeButton(mContext.getString(R.string.available_timer_cancel), new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int id) {
+                        handler.removeCallbacksAndMessages(null);
+                        dialog.cancel();
+                    }
+                });
+        //add message to load machine before
+        LayoutInflater inflater = LayoutInflater.from(mContext);
+        View q = inflater.inflate(R.layout.view_available_machine, null);
+        machineWaitingDialog.setView(q);
+        TextView number = (TextView) q.findViewById(R.id.machine_name_number);
+        number.setText(getNumberFromName(m));
+        final AlertDialog alertDialog = machineWaitingDialog.create();
+        alertDialog.show();
+        alertDialog.setOnCancelListener(new DialogInterface.OnCancelListener() {
+            @Override
+            public void onCancel(DialogInterface dialog) {
+                lockListener.onUnlock();
+            }
+        });
+        alertDialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener(new View.OnClickListener() {
```

Comment:
I think you can attach these listeners in the Builder, and I think that would be cleaner than doing it here after the fact.